### ISSUE TITLE
Bible book chooser improvements

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#332]
- Remove gradient (check if it looks good)
- Transpose books #142 
- High contrast indicator for currently selected book in book chooser menu #97
- Ideas #715
- Optional **quick one-view chooser**(s), similar to what is in MySword (see screenshot)

![Screenshot_20210323-124052](https://user-images.githubusercontent.com/5811789/112134218-2af61f80-8bd5-11eb-80a2-18dbad5b1e55.png)

**Comment by tuomas2** [comment]
There was typo in commit msg.. (4a8771c)

**Comment by tuomas2** [comment]
From darryl:

>  Please consider indicating the current book, in the Book Selector, in INVERSE colours. Eg: when Matthew (usually Orange text on black gradient) becomes current, then it should be Black text on Orange background; Romans would be Black text on yellow background (when current); etc.

**Comment by tuomas2** [comment]
User via Google play:
> I love this bible so much, although I will shoot you mails, I am having problems with the choose book interface. Could you give us options? Maybe the old testament scrollable on one side and the new testament scrollable on the other. Then without the colours, just on a transparent white background to ease quick recognition?

**Comment by Screwtapello** [comment]
Currently, the Bible book chooser shows the abbreviation for each book (Job, Joe, Joh, Jon, etc.) and when you press a particular button, a tooltip appears above the button so you can see which button is obscured by your finger.

This is great, but I think it would be even better if the tooltip showed the long name of the book, as confirmation I'm pressing the correct button. I understand the buttons themselves have to show the abbreviation due to space constraints, but the tooltip can take much more space, and some of those abbreviations are confusingly terse.

- [ ] Showing the full name of the book would mean a new Christian has one less thing they need to memorise in order to find their way around the Bible.

**Comment by cmahte** [comment]
Including both the interface abbreviation and a title from within the text currently displayed adds value beyond "new christian." 

Abbr (TITLE in the Current work) would be perfect.  not only display the available interface translation of the abbr, but a title in the language of the text currently displayed. So not just a 'newbie' feature. And for folks like me... helps to confirm the physical book i'm comparing a digital reference to... that  I've interpreted the header text on the physical page correctly in choosing my lookup. 

As opposed to having to look in chapter 1 first for the language  of the title match... 

If this opens up for edit.... just a thought.

**Comment by timbze** [comment]
From someone with sight impairment:

> I have found another difficulty – the abbreviation of the books in small writing in a 'multi-button' layout.  Can they be rearranged in two columns in bigger writing please?? – again a light / lime green background with bold text would be fabulous.

**Comment by onomou** [comment]
> Currently, the Bible book chooser shows the abbreviation for each book (Job, Joe, Joh, Jon, etc.) and when you press a particular button, a tooltip appears above the button so you can see which button is obscured by your finger.
> 
> This is great, but I think it would be even better if the tooltip showed the long name of the book, as confirmation I'm pressing the correct button. I understand the buttons themselves have to show the abbreviation due to space constraints, but the tooltip can take much more space, and some of those abbreviations are confusingly terse.
> 
> Showing the full name of the book would mean a new Christian has one less thing they need to memorise in order to find their way around the Bible.

I was just trying my hand at understanding how the code is put together and made a patch to show the full name of the book. If it is useful and done properly, I would love to have it pulled in. The original issue's screenshot would be the best book/chapter selector IMO, but this at least patches up the current view a bit.

**Comment by agrogers** [comment]
I like the current selector - it seems almost as efficient as it can be - and it is definitely better with books going across not down. But sometimes efficiency requires more mental power than something that is easier on the brain but a little slower - like Windows verse Linux perhaps :) Certainly something to look at. But

I suggest we delay this one to after v3.4 release

**Comment by tuomas2** [comment]
I have prepared strings (chooser_*) for the optional new selector so it can be implemented.

**Comment by agrogers** [comment]
> From darryl:
> 
> > Please consider indicating the current book, in the Book Selector, in INVERSE colours. Eg: when Matthew (usually Orange text on black gradient) becomes current, then it should be Black text on Orange background; Romans would be Black text on yellow background (when current); etc.

This is done now, Darryl. Not sure when it will get into a release though.
See https://github.com/AndBible/and-bible/pull/2179

**Comment by agrogers** [comment]
@onomou 

> Currently, the Bible book chooser shows the abbreviation for each book (Job, Joe, Joh, Jon, etc.) and when you press a particular button, a tooltip appears above the button so you can see which button is obscured by your finger.

If you hold down the button the fullname pops up. I never use it but it is kind of cool.

**Comment by onomou** [comment]
I can confirm that newer version of AndBible show the full book name when you hold down the button. Thanks.

**Comment by agrogers** [comment]
> Currently, the Bible book chooser shows the abbreviation for each book (Job, Joe, Joh, Jon, etc.) and when you press a particular button, a tooltip appears above the button so you can see which button is obscured by your finger.
> 
> This is great, but I think it would be even better if the tooltip showed the long name of the book, as confirmation I'm pressing the correct button. I understand the buttons themselves have to show the abbreviation due to space constraints, but the tooltip can take much more space, and some of those abbreviations are confusingly terse.


I have added this functionality now. You can see what it looks like here: https://github.com/AndBible/and-bible/pull/2193

**Comment by tuomas2** [comment]
@meichthys why was this closed?

**Comment by meichthys** [comment]
Sorry, it looked like it had been completed and merged here: https://github.com/AndBible/and-bible/pull/2193

**Comment by meichthys** [comment]
Now I see the one-view chooser idea hadn't been implemented yet. I do like that idea, but also don't mind the existing chooser. I'll think about this a bit and follow up on the other discussion thread with any ideas. #715

**Comment by tuomas2** [comment]
We could have different choosers, both probably have pros and cons.

**Comment by meichthys** [comment]
I do have a chooser method in mind of my own. It would look something like three sliders (horizontal or vertical). The user would slide their finger (red dotted path) across the book slider until they found the book they wanted, then drag their finger down to the chapter slider and choose the chapter and then drag their finger to the verse selector and choose the verse - all while the top reference updates in real time. Ideally this could all be done in a single swipe, however if it required three separate swipes on each of the sliders that would probably be fine too.


![Screenshot_20250429-063722.png](https://github.com/user-attachments/assets/bdc5bb0c-f724-48d1-8c87-c835c6610499)

**Comment by tuomas2** [comment]
Looks interesting idea. I'm not sure how accurate that would be but perhaps something that could be tried out.

**Comment by meichthys** [comment]
Yeah, I have not actually tried it myself and it may not be very precice for books that have so many chapters like Psalms. ( Perhaps the vertical sliders would be better since it would give more screen real estate for the sliders.)
If I get a chance, I will try to make a mock-up of this in a Python program.

**Comment by meichthys** [comment]
Here's an example of the horizontal sliders. It seems to work reasonably well. The example below does require you to lift your finger for each slider, but it gives a bit of the feel at least (I've included a zipped up apk if you'd like to try it out):

https://github.com/user-attachments/assets/a0bf1fb7-706a-4484-a4fe-49b500dc9ca7

[app-debug.zip](https://github.com/user-attachments/files/19996761/app-debug.zip)

**Comment by tuomas2** [comment]
Interesting. It would require memorizing the book order, but as an optional mode it would be nice addition.
